### ISSUE TITLE
No changelog in apt repo

### ISSUE THREAD

**Issue by se4n01** [mattermost/mattermost#25058]
Mattermost does not include a changelog in its repo, this would be convenient to have.

Check for a changelog in the repo:

```
$ sudo apt-get changelog mattermost
E: Failed to fetch changelog:/mattermost.changelog  Changelog unavailable for mattermost=9.1.0-0
```

Thanks and best wishes,
Sean

**Comment by amyblais** [comment]
Hi @se4n01, our changelog is located here https://docs.mattermost.com/install/self-managed-changelog.html.

**Comment by amyblais** [comment]
The GitHub file is here: https://github.com/mattermost/mattermost/blob/master/CHANGELOG.md.

**Comment by amyblais** [comment]
Hi @se4n01, let me know if the links above are helpful? If not, let me know more details on what we can improve here.

**Comment by se4n01** [comment]
Hello and thanks for the reply.

This is not really helpful. I know the location of the changelog. This is a Feature Request to enhance the apt repos you provide located at `deb.packages.mattermost.com`

In linux, when you use apt to manage your packages you can see the changelog of an individual package on the machine by using

`apt changelog <package>` 

This is what does not work in the debian repos you provide.

To make it work, you need to do the following:

When building, make sure you have a changelog file in the debian directory structured in the normal way:
```
mattermost (version) distribution; urgency=urgency
  * Change details.

-- Maintainer Name <email>  Date
```

Then when you run debian-buildpackage or use deb_helper or whatever your process is, make sure you also move the .changes file it generates to the debian root on the server that hosts your apt repo. You also need to make sure the `Packages` file in your debian repo contains a link to the changelog like `Changelog: https://deb.packages.mattermost.com/debian/jammy-64/mattermost.changelog.jammy.gz`

At that point after you sign your packages and whatever else you are doing to your repo, and in the end `apt changelog mattermost` will work and display the changelog.

tl;dr this is a feature request about enhancing your apt repository structure.

Best wishes,

Sean

**Comment by amyblais** [comment]
Thank you for the details, I've opened a ticket for our team to discuss.

**Comment by amyblais** [comment]
Thanks, appreciate your feedback @se4n01,

We currently don't have capacity to work on this internally. Would you like to [contribute this in the feature idea forum](https://mattermost.uservoice.com/forums/306457-general/) so it can be discussed, upvoted and considered for a [help wanted ticket](https://docs.mattermost.com/process/help-wanted.html)?

Please include a link back to this GitHub issue. If you're interested in implementing, please say so and we'll prioritize the review. 

You get **10** votes in the feature idea forum, and each one influences the future of the project.

**Comment by amyblais** [comment]
Closing as this can be tracked on UserVoice,
